trk/src/api/TapsApi.js

```diff
@@ -44,50 +36,34 @@ function TapsApi() {
-                    await updateTap(nextTaps.docs[0].id, { isSessionStart: true, sessionImages: (oldTap._data.sessionImages !== undefined? oldTap._data.sessionImages: []), isSelected: (oldTap._data.isSelected !== undefined? oldTap._data.isSelected: true), sessionTitle: (oldTap._data.sessionTitle!== undefined? oldTap._data.sessionTitle: ''), tagged: (oldTap._data.tagged!== undefined? oldTap._data.tagged: [])});
-                    const scheduleFunction = firebaseFunctions.functions().httpsCallable('scheduleFunction');
-                    let expiryTimeForFunction;
-                    if (oldTap._data.expiryTime instanceof firebase.firestore.Timestamp) {
-                        // Convert Firebase Timestamp to JavaScript Date object
-                        expiryTimeForFunction = oldTap._data.expiryTime.toDate().toISOString();
-                    } else if (oldTap._data.expiryTime instanceof Date) {
-                        // Already a JavaScript Date object
-                        expiryTimeForFunction = oldTap._data.expiryTime.toISOString();
-                    } else {
-                        // If it's not a Firebase Timestamp or JavaScript Date, use it as is
-                        expiryTimeForFunction = oldTap._data.expiryTime;
-                    }
-
-                    scheduleFunction({ tapId: nextTaps.docs[0].id, expiryTime: expiryTimeForFunction })
-                        .then((result) => {
-                            //console.log('Function result:', result.data);
-                        }).catch((error) => {
-                            console.error('Error calling function:', error);
-                        });
 
+            //get the session that contained the tap
+            const sessionSnapshot = await SessionsApi().getSessionByTap(tapId);
+            console.log('The fetched session is: ', sessionSnapshot.docs);
+            if (!sessionSnapshot.empty) {
+                const sessionObject = sessionSnapshot.docs[0].data();
+                let climbsArray = sessionObject.climbs;
+                let featuredClimbCopy = sessionObject.featuredClimb;
+                const tapIdIndex = climbsArray.indexOf(tapId);
+                if (tapIdIndex > -1) {
+                    climbsArray.splice(tapIdIndex, 1);
+                }
+                if (featuredClimbCopy === tapId) {
+                    featuredClimbCopy = climbsArray.length > 0 ? climbsArray[0] : ''; // Set to first element or null if climbsArray is empty
+                }
+                if (climbsArray.length == 0) {
+                    //session is empty, archive it
+                    await SessionsApi().updateSession(sessionSnapshot.docs[0].id, {archived: true});
+                } else {
+                    //update session with the new climbsarray and featuredClimb
+                    await SessionsApi().updateSession(sessionSnapshot.docs[0].id, {climbs: climbsArray, featuredClimb: featuredClimbCopy});
```

Comment:
amazing

Reply:
good job covering edge cases (array length check & assignment to archived if empty)